### ISSUE TITLE
Some Issues on Conditional Diffusion

### ISSUE THREAD

**Issue by ghost** [sail-sg/MDT#19]
Hello！Thank you for your excellent work！

May I ask how to correctly modify the code section for conditional diffusion? After trying to modify it myself, I found that there was an abnormal phenomenon in the loss.

May I ask if the correct approach is to modify it in _**masked_diffusion/gaussian_diffusion. py**_?

**Comment by gasvn** [comment]
What kind of conditions do you want to modify?

**Comment by ghost** [comment]
> What kind of conditions do you want to modify?

@gasvn 
I hope this model can achieve rain removal for rainy image. In the conditional diffusion, the rainy image is generally used directly as the conditional image. First, the corresponding non-rainy image is q_sample() to obtain x_t, and then x_t and the rainy image as the condition are **_torch.cat (x_t, x_rain, dim=1)_** and then input into the network. 

For this paper, I also made similar modifications and the loss occurred during the training process and the generated image was completely wrong. Is this modification reasonable in this work?

**Comment by Feynman1999** [comment]
> > What kind of conditions do you want to modify?
> 
> @gasvn I hope this model can achieve rain removal for rainy image. In the conditional diffusion, the rainy image is generally used directly as the conditional image. First, the corresponding non-rainy image is q_sample() to obtain x_t, and then x_t and the rainy image as the condition are **_torch.cat (x_t, x_rain, dim=1)_** and then input into the network.
> 
> For this paper, I also made similar modifications and the loss occurred during the training process and the generated image was completely wrong. Is this modification reasonable in this work?

any update on this problem?
